### PR TITLE
Pull event address from possible branch node. DDFFORM-402

### PR DESCRIPTION
**Pull request by rasben**
Pull event address fallback from possible branch node. DDFFORM-402

Making a custom field formatter, that looks up if a branch node
is associated with the field, and sets the branch address as fallback
if nothing exists on the event already.

### CODE REVIEW

**Review by kasperg — CHANGES_REQUESTED** [review]
Please reconsider the logic in this PR-

Bascally I think you reversed the business intention. If an address is shown for the event then it should be shown. Otherwise it should fall back to the address of the related branch.

With the current approach it is not possible to have an event which is both related to a branch and with a custom address.

**Comment by rasben** [comment]
@kasperg You were right - it was the other way around.

I've updated the PR now, along with adding example content.
I've also made sure that the branch is displayed on the event view.

File: config/sync/field.field.eventinstance.default.field_branch.yml
Comment (kasperg) [inline config/sync/field.field.eventinstance.default.field_branch.yml:14]: Please reconsider these descriptions.

I think they are leftovers from the previous implementation where the logic was reversed.

This also applies for the three other descriptions.

File: phpstan-baseline.neon
Comment (kasperg) [inline phpstan-baseline.neon:12]: Please move this to an `ignoreErrors` entry in `phpstan.neon.dist`.

I this it is perfectly acceptable that you ignore such an error related to Drupals render arrays but ignoring it here will leave the next guy with the same hard-to-fix error when implementing a field formatter.

I think something like this should do the trick:

```
    # Field formatters return render arrays which we cannot provide more detailed typing of.
    - '#Method Drupal\\.*\\Plugin\\Field\\FieldFormatter\\.*\:\:viewElements\(\) return type has no value type specified in iterable type array\.#'
```

File: web/modules/custom/dpl_event/src/Plugin/Field/FieldFormatter/BranchAddressFormatter.php
Comment (kasperg) [inline web/modules/custom/dpl_event/src/Plugin/Field/FieldFormatter/BranchAddressFormatter.php:31]: Please reconsider the context.

If in doubt use the module name.

File: web/modules/custom/dpl_event/src/Plugin/Field/FieldFormatter/BranchAddressFormatter.php
Comment (kasperg) [inline web/modules/custom/dpl_event/src/Plugin/Field/FieldFormatter/BranchAddressFormatter.php:24]: :+1: Well done implementing this as a field formatter.

Personally I like this approach compared to an endless amount of preprocessing.

File: web/modules/custom/dpl_example_content/content/eventseries/2ce9d9b0-7f10-4a17-9a5e-383d36c1ed93.yml
Comment (kasperg) [inline web/modules/custom/dpl_example_content/content/eventseries/2ce9d9b0-7f10-4a17-9a5e-383d36c1ed93.yml:107]: :+1: for remembering to create/update example content with new features as we add them.

**Review by kasperg — APPROVED** [review]
:+1: LGTM

A few things needs a little cleanup but I do not think the changes need another review before merging.

**Pull request merged**
